### PR TITLE
introduce WAR_URL argument so one can use alternate download URL

### PR DESCRIPTION
**Pull request by ndeloof**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -38,11 +38,12 @@ ARG JENKINS_VERSION
 ENV JENKINS_VERSION ${JENKINS_VERSION:-2.7.1}
 ARG JENKINS_SHA
 ENV JENKINS_SHA ${JENKINS_SHA:-12d820574c8f586f7d441986dd53bcfe72b95453}
+ARG WAR_URL
+ENV WAR_URL ${WAR_URL:-http://repo.jenkins-ci.org/public/org/jenkins-ci/main/jenkins-war/${JENKINS_VERSION}/jenkins-war-${JENKINS_VERSION}.war}
```
Comment (carlossg) [inline Dockerfile]: do we need the ENV ?

File: Dockerfile [hunk 2]
```diff
@@ -34,17 +34,20 @@ RUN curl -fsSL https://github.com/krallin/tini/releases/download/v${TINI_VERSION
...
 ENV JENKINS_VERSION ${JENKINS_VERSION:-2.7.1}
+
+# jenkins.war checksum, download will be validated using it
 ARG JENKINS_SHA
-ENV JENKINS_SHA ${JENKINS_SHA:-12d820574c8f586f7d441986dd53bcfe72b95453}
-ARG WAR_URL
-ENV WAR_URL ${WAR_URL:-http://repo.jenkins-ci.org/public/org/jenkins-ci/main/jenkins-war/${JENKINS_VERSION}/jenkins-war-${JENKINS_VERSION}.war}
+
+# Can be used to customize where jenkins.war get downloaded from
+ARG JENKINS_URL
```
Comment (carlossg) [inline Dockerfile]: Seems easier to read as

    ARG JENKINS_SHA=12d820574c8f586f7d441986dd53bcfe72b95453
    ARG JENKINS_URL=http://repo.jenkins-ci.org/public/org/jenkins-ci/main/jenkins-war/${JENKINS_VERSION}/jenkins-war-${JENKINS_VERSION}.war

Comment (ndeloof) [inline Dockerfile]: good point

File: Dockerfile [hunk 3]
```diff
@@ -34,16 +34,20 @@ RUN curl -fsSL https://github.com/krallin/tini/releases/download/v${TINI_VERSION
...
-# could use ADD but this one does not check Last-Modified header 
+# Can be used to customize where jenkins.war get downloaded from
+ARG JENKINS_URL=http://repo.jenkins-ci.org/public/org/jenkins-ci/main/jenkins-war/${JENKINS_VERSION}/jenkins-war-${JENKINS_VERSION}.war
+
+# could use ADD but this one does not check Last-Modified header neither does it allow to control checksum 
 # see https://github.com/docker/docker/issues/8331
-RUN curl -fsSL http://repo.jenkins-ci.org/public/org/jenkins-ci/main/jenkins-war/${JENKINS_VERSION}/jenkins-war-${JENKINS_VERSION}.war -o /usr/share/jenkins/jenkins.war \
-  && echo "$JENKINS_SHA  /usr/share/jenkins/jenkins.war" | sha1sum -c -
+RUN curl -fsSL ${JENKINS_URL} -o /usr/share/jenkins/jenkins.war \
+  && echo "${JENKINS_SHA} /usr/share/jenkins/jenkins.war" | sha1sum -c -
```
Comment (carlossg) [inline Dockerfile]: the extra space (2) between sha and file is needed in alpine

Comment (ndeloof) [inline Dockerfile]: :)

**Pull request merged**